local/include/core/proxy_protocol_util.h

```diff
@@ -0,0 +1,118 @@
+/** @file
+ * Common data structures and definitions for Proxy Verifier tools.
+ *
+ * Copyright 2023, Verizon Media
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "swoc/Errata.h"
+#include "swoc/BufferWriter.h"
+#include "swoc/TextView.h"
+#include "swoc/swoc_ip.h"
+
+// PROXY header version
+enum class ProxyProtocolVersion { NONE = 0, V1 = 1, V2 = 2 };
+
+// PROXY header v1 end of header.
+static constexpr swoc::TextView PROXY_V1_EOH{"\r\n"};
+
+// The maximum size of a PROXY header(v1 and v2) without the TLV support.  This
+// is used to specify the data size to peek from the socket.
+static constexpr size_t MAX_PP_HDR_SIZE = 108;
+
+// V1 and V2 header signatures
+using namespace std::literals;
+static const swoc::TextView V1SIG("PROXY");
+constexpr swoc::TextView V2SIG = "\x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A"sv;
+
+static constexpr char PP_V1_DELIMITER = ' ';
+
+// This is the union taken from the RFC that specifies v1 and v2 PROXY header.
+// Note that the linux socket address is removed from the v2 header, as it is
+// not supported in Proxy Verifier.
+union ProxyHdr {
```

Comment:
There's no official RFC for Proxy Protocol, but this is the closest thing to it:
https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt

Let's add a reference for that.

Reply:
added the reference.